### PR TITLE
ktlint formatter added and resolved findings

### PR DESCRIPTION
**Pull request by idperez**
Adds a formatter to the project and addresses formatting findings. Readme updated as well.

Using the Spotless project (https://github.com/diffplug/spotless/tree/master/plugin-gradle) which has ktlint (https://ktlint.github.io/) support and license header enforcement. 

Running `gradle build` will run the formatter. Any errors can be resolved by running `gradle spotlessApply`.

### CODE REVIEW

File: test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt [hunk 1]
```diff
@@ -30,6 +28,9 @@ import org.codice.security.sign.Decoder.InflationException.InflErrorCode
 import org.w3c.dom.Node
 import java.io.IOException
 
+/* ktlint-disable no-wildcard-imports */
+import org.codice.compliance.SAMLSpecRefMessage.*
+
```
Comment (blen-desta) [inline test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: ❓ Are we not able to import what we need without using wildcard for the `SAMLSpecRefMessage` stuff?

File: test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt [hunk 2]
```diff
@@ -35,6 +34,9 @@ import org.codice.security.sign.Encoder
 import org.codice.security.sign.SimpleSign
 import java.time.Instant
 
+/* ktlint-disable no-wildcard-imports */
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.*
```
Comment (blen-desta) [inline test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt]: ✏️ We should import the individual `SSOConstants` instead of `suppressing?` this.

Comment (idperez) [reply test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt]: Could the number of SSOConstants used here grow? Importing each one could get out of hand

File: test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt (hunk 1 above)
Comment (idperez) [reply test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: Wildcard imports are not allowed under ktlint so we need a suppressor

Comment (coyotesqrl) [reply test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: Wildcards are disallowed in our Java standard; let's be consistent.

https://google.github.io/styleguide/javaguide.html#s3.3-import-statements

File: test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt (hunk 2 above)
Comment (blen-desta) [reply test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt]: I don't think it will. As of right now you just need four. `SAML_REQUEST`, `SIG_ALG`, `SIGNATURE` and `RELAY_STATE`

File: test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt (hunk 1 above)
Comment (blen-desta) [reply test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: Yeah, that was a question more for @brjeter on whether or not we can have individual import instead of suppressing it. If doing the individual is too much, I'm okay with the suppression.

File: test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt (hunk 2 above)
Comment (idperez) [reply test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt]: Okay, I would say in this case we can import each one. Some of the files it may be overkill

File: test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt (hunk 1 above)
Comment (blen-desta) [reply test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: If it's too many one thing you can do is import `org.codice.compliance.SAMLSpecRefMessage` instead and of using `SAMLCore_3_2_2_a` in the code do `SAMLSpecRefMessage.SAMLCore_3_2_2_a`

**Comment by coyotesqrl** [comment]
Honestly, it's basically invisible to anyone using a modern ide; the
imports block is collapsed.

**Comment by coyotesqrl** [comment]
One last thought on the imports...if we did decide that the imports for consts/enums became too long, we could as @blen-desta suggested, just import `org.codice.compliance.SAMLSpecRefMessage` and then create a [type alias](https://kotlinlang.org/docs/reference/type-aliases.html) to `SAMLSpecRefMessage` to make the references to the enums shorter, something like:
```kotlin
import org.codice.compliance.SAMLSpecRefMessage

typealias SpecMsg = SAMLSpecRefMessage
SpecMsg.SAMLCore_3_2_2_a
```
But again, I don't believe it really matters if the import block grows long.

File: test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt (hunk 2 above)
Comment (brjeter) [reply test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt]: I agree with both of you. In this instance, we can import the individual ones but in some edge cases it may be better to suppress.

File: test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt (hunk 1 above)
Comment (brjeter) [reply test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt]: I'd rather not do that ☝️ I'm fine with the suppression or, to Richard's point, just importing each individual message. Developers rarely look at the import blocks.

File: build.gradle [hunk 3]
```diff
@@ -62,6 +64,14 @@ subprojects {
         compile "org.slf4j:slf4j-api:$org_slf4j_version"
         compile "com.google.guava:guava:$guava_version"
     }
+
+    spotless {
+        kotlin {
+            ktlint()
+
+            licenseHeaderFile rootProject.file('spotless.license.kt')
```
Comment (brjeter) [inline build.gradle]: ✏️ Shouldn't this be renamed to the "codice license" or something?
